lib/simple_forms_api/form_remediation.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module SimpleFormsApi
+  module FormRemediation
+    class Error < StandardError
+      attr_reader :base_error, :details
+
+      DEFAULT_MESSAGE = 'An error occurred during the form remediation process'
+
+      def initialize(message: DEFAULT_MESSAGE, error: nil, details: nil, backtrace: nil)
+        super(message)
+        @base_error = error
+        @details = details
+        @custom_backtrace = backtrace
+      end
+
+      def message
+        [super, custom_message].compact.join(' - ')
+      end
+
+      def backtrace
+        @custom_backtrace || base_error&.backtrace || super
```

Comment:
How do we use the `backtrace`? As in, where does it appear or how do we get to look at it?

Reply:
Usually, you'd have to access it via IRB or PRY or some other way. We can include the backtrace in the error logging though which might prove useful for debugging purposes.